### DISCUSSION TITLE
How to set Tif Tag PHOTOMETRIC form BlackIsZero to WhiteIsZero ?

### DISCUSSION THREAD

**Discussion by Charltsing** [dlemstra/Magick.NET#1760] [Help]
### Magick.NET version

13.10.0

### Environment (Operating system, version and so on)

Windows10, net framework 4.7.2

### Description

I save pbm to tif,  Tif Tag PHOTOMETRIC is BlackIsZero.
I want to change BlackIsZero to WhiteIsZero to align with the behavior of other software

![BlackIsZero](https://github.com/user-attachments/assets/128b9f92-10ce-44e1-8bd5-a9006f0b0bf6)

[BlackIsZero.zip](https://github.com/user-attachments/files/17895339/BlackIsZero.zip)

```
using (ImageMagick.MagickImage processed = new ImageMagick.MagickImage(pnmBytes))
{
    processed.Format = ImageMagick.MagickFormat.Tif;

    processed.Settings.Depth = 1;
    processed.Settings.ColorSpace = ImageMagick.ColorSpace.Gray;
    processed.Settings.ColorType = ImageMagick.ColorType.Bilevel;
    processed.Settings.Compression = ImageMagick.CompressionMethod.Group4;
   
    // How to set  Tif Tag PHOTOMETRIC to WiiteIsZero

    processed.Write(outstream, ImageMagick.MagickFormat.Tif);
}
```

**Comment by dlemstra** [comment]
What happens when you add this:

 ```C#
processed.Settings.SetDefine(MagickFormat.Tiff, "quantum:polarity", "min-is-white");
```

This should probably be added to the `TiffWriteDefines`.

**Comment by Charltsing** [comment]
it does not work !

I creat a tifdemo project to test BlackIsZero, use MetadataExtractor to Read Tags
[TifDemo.zip](https://github.com/user-attachments/files/17928183/TifDemo.zip)

![2](https://github.com/user-attachments/assets/6436a0ab-6eaf-45ee-a4c1-56865e569fa8)


```
using (MagickImage processed = new MagickImage(fullpath))
            {
                processed.Format = MagickFormat.Tiff;

                processed.Settings.Depth = 1;
                processed.Settings.ColorSpace = ColorSpace.Gray;
                processed.Settings.ColorType = ColorType.Bilevel;
                processed.Settings.Compression = CompressionMethod.Group4;

                processed.Settings.SetDefine(MagickFormat.Tiff, "quantum:polarity", "min-is-white");

                processed.Write(outpath, MagickFormat.Tiff);
            }
```

**Comment by dlemstra** [comment]
It should be `processed.Settings.SetDefine("quantum:polarity", "min-is-white");` instead. Thought this was a tiff specific setting.

**Comment by Charltsing** [comment]
it work fine
![3](https://github.com/user-attachments/assets/f954616f-1639-4f5c-ab6f-b6818e3ae87d)

thanks a lot
